Various fixes for Rcmdcheck

dfalster opened the pull request:
Rcmdcheck complains about lots of issues. For example, see mentions at

https://github.com/traitecoevo/plant/actions/runs/3263550975/jobs/5362669197

or rund `rcmdcheck::rcmdcheck()` locally

These fixes try to rectify

Locally I'm still getting a fail, possibly due to request for specific version of `tidyselect`

I only get the fail when running tests via `rcmdcheck::rcmdcheck()` and not via `devtools::test()`. Worse am yet to isolate the cause. I've tried reverting functions in the `R/tidy_outputs.R` but it was still occurring. 

@aornugent Can you run `rcmdcheck::rcmdcheck()` at your end and let me know how it goes?

dfalster committed:
Variopus fixes for Rcmdcheck

aornugent commented:
Hi - these are my local results:

```
> rcmdcheck::rcmdcheck()
── R CMD build ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
pdflatex not found! Not building PDF manual.
✔  checking for file ‘.../DESCRIPTION’ ...
─  preparing ‘plant’: (6.5s)
✔  checking DESCRIPTION meta-information ...
─  cleaning src
─  installing the package to build vignettes
✔  creating vignettes (2m 8.3s)
─  cleaning src
─  checking for LF line-endings in source and make files and shell scripts (1.1s)
─  checking for empty or unneeded directories
─  building ‘plant_2.0.0.9000.tar.gz’
   
── R CMD check ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
─  using log directory ‘/tmp/Rtmpfu2yW2/file4e39596bde22/plant.Rcheck’
─  using R version 4.2.1 (2022-06-23)
─  using platform: x86_64-pc-linux-gnu (64-bit)
─  using session charset: UTF-8
✔  checking for file ‘plant/DESCRIPTION’ ...
─  this is package ‘plant’ version ‘2.0.0.9000’
─  package encoding: UTF-8
✔  checking package namespace information ...
✔  checking package dependencies (3.3s)
✔  checking if this is a source package ...
✔  checking if there is a namespace
✔  checking for executable files (489ms)
✔  checking for hidden files and directories
✔  checking for portable file names ...
✔  checking for sufficient/correct file permissions
✔  checking whether package ‘plant’ can be installed (2m 3s)
N  checking installed package size ...
     installed size is 49.3Mb
     sub-directories of 1Mb or more:
       R      2.3Mb
       docs   2.4Mb
       libs  43.8Mb
✔  checking package directory ...
✔  checking ‘build’ directory
✔  checking DESCRIPTION meta-information ...
✔  checking top-level files ...
✔  checking for left-over files
✔  checking index information ...
✔  checking package subdirectories ...
✔  checking R files for non-ASCII characters ...
✔  checking R files for syntax errors ...
✔  checking whether the package can be loaded ...
✔  checking whether the package can be loaded with stated dependencies ...
✔  checking whether the package can be unloaded cleanly ...
✔  checking whether the namespace can be loaded with stated dependencies ...
✔  checking whether the namespace can be unloaded cleanly ...
✔  checking loading without being on the library search path ...
✔  checking dependencies in R code (965ms)
✔  checking S3 generic/method consistency (637ms)
✔  checking replacement functions ...
✔  checking foreign function calls (671ms)
/  checking R code for possible problems ...
```

dfalster committed:
Resolve failing tests

dfalster committed:
Put slowest test last

aornugent commented:
I'm not quite sure where this got up to, but I think we want to resolve in favour of the `pass_checks ` branch?

```r
#' @export
#'
#' @importFrom rlang .data
integrate_over_size_distribution <- function(tidy_species_data) {
  tidy_species_data  %>%
    dplyr::select(-.data$node) %>% stats::na.omit() %>%
    dplyr::filter(.data$step > 1) %>% 
    dplyr::group_by(.data$step, .data$time, .data$patch_density, .data$species) %>% 
    dplyr::summarise(

<<<<<<< pass_checks

      density_integrated = -trapezium(.data$height, .data$density), 
      min_height = min(.data$height),
      dplyr::across(tidyselect::where(is.double) & !c(.data$density, .data$density_integrated, .data$min_height) , ~-trapezium(height, density * .x)), 

=======

      density_integrated = -trapezium(height, density), 
      min_height = min(height),
      dplyr::across(where(is.double) & !c(density, density_integrated, min_height), 
                    ~-trapezium(height, density * .x)), 

>>>>>>> develop

      .groups = "drop"
    ) %>% 
    dplyr::rename(density = .data$density_integrated)
}
```

dfalster committed:
Relax strictness of some tests

- expect_equal allows for small numerical differences, expect_identicalis stricter

dfalster committed:
reframe is new and preferred over summarise when multiple rows per group

dfalster committed:
Merge branch 'fix-tests' into pass_checks

dfalster committed:
Resolving further issues

dfalster committed:
More leniency on test
